core/src/test/java/cz/xtf/core/openshift/mocked/smoke/BasicOpenShiftTest.java

```diff
@@ -0,0 +1,102 @@
+package cz.xtf.core.openshift.mocked.smoke;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import cz.xtf.core.openshift.OpenShift;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.fabric8.openshift.client.server.mock.OpenShiftServer;
+
+public class BasicOpenShiftTest {
+
+    private static final String TEST_RESOURCE_LABEL_NAME_APP = "app";
+    private static final String TEST_RESOURCE_LABEL_VALUE_APP = "xtf-core-test-openshift-mocked-smoke";
+    private final static Map<String, String> TEST_RESOURCE_LABELS = Collections.singletonMap(
+            TEST_RESOURCE_LABEL_NAME_APP, TEST_RESOURCE_LABEL_VALUE_APP);
+
+    private static final String TEST_CONFIGMAP_NAME = "test-configmap";
+    private static final String TEST_DEPLOYMENT_NAME = "test-deployment";
+
+    private final OpenShiftServer openShiftServer;
+    private final OpenShift openShiftClient;
+
+    public BasicOpenShiftTest() {
+        this.openShiftServer = new OpenShiftServer(false, true);
+        this.openShiftServer.before();
+        // we want to test the XTF OpenShift client, but we need to configure it with the mocked server client
+        // properties
+        OpenShiftClient mockedServerClient = openShiftServer.getOpenshiftClient();
+        this.openShiftClient = OpenShift.get(
```

Comment:
I think we'll need some level of abstraction here which will decide whether to use mock or real Openshift cluster. wdyt?

Reply:
+1, thanks @mnovak1 - I'll add something.

Reply:
Hi @mnovak1 - while thinking about this, I still don't have a strong opinion on the type of criteria we should use to drive this. A XTF property, e.g.: `xtf.test.server.type=(actual|mocked)`? Or a Maven profile? Just brainstorming here.

WDYT?

CC @simkam

Reply:
I prefer to use only mock server in unit tests and (if needed) create a new maven module with integration tests running on OpenShift.

Reply:
Ah, actually yes, that could be a solution that BTW would allow for driving the two things separately and asynchronously. WDYT @mnovak1 ?

Reply:
Agreed. I think it's practical solution as running against real instance will be really rare. Also developing tests for both (moc and real openshift) and keep them stable would be hard achieve.